frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/advanced-section.tsx

```diff
@@ -1,48 +1,61 @@
 import * as React from 'react';
 import { useTranslation } from 'react-i18next';
 import {
-  Checkbox,
-  CheckboxProps,
   ExpandableSection,
   ExpandableSectionProps,
   FormGroup,
+  Select,
+  SelectOption,
+  SelectProps,
+  SelectVariant,
 } from '@patternfly/react-core';
 import { DeploymentType } from '../../../../constants/create-storage-system';
 import { WizardDispatch, WizardState } from '../../reducer';
 import './backing-storage-step.scss';
 
+const options = [DeploymentType.FULL, DeploymentType.MCG];
+
 export const AdvancedSection: React.FC<AdvancedSelectionProps> = ({
   deployment,
   isAdvancedOpen,
   dispatch,
 }) => {
   const { t } = useTranslation();
+  const [isSelectOpen, setIsSelectOpen] = React.useState(false);
 
-  const handleSelection: CheckboxProps['onChange'] = (checked) => {
+  const handleSelection: SelectProps['onSelect'] = (_, value) => {
     dispatch({
       type: 'backingStorage/setDeployment',
-      payload: checked ? DeploymentType.MCG : DeploymentType.ALL,
+      payload: value as DeploymentType,
```

Comment:
On `SelectProps['onSelect']` the value is string hence not matching with payload which is of "DeploymentType"